web/gridt/tests/integration/resources/test_user.py

```diff
@@ -23,9 +30,214 @@ def test_update_bio(self):
+    def test_send_reset_password_email_wrong(self):
+        with self.app_context():
+            # Make a request with nonexistent e-mail
+            email = "nonexistent@email.com"
+            
+            resp = self.client.post(
+                "/request_password_reset",
+                json={
+                    "email": email
+                },
+            )
+
+            # Test that email will not get sent
+
+
+            # In order to not give away sensitive information
+            # message must be the same as a successful attempt
+            # self.assertIn("message", resp.get_json())
+            # self.assertEqual(resp.status_code, 200)
+
+
+    def test_send_reset_password_email_correct(self):
+        # Request reset password, e-mail in database
+        with self.app_context():
+            user = self.create_user()
+
+            resp = self.client.post(
+                "/request_password_reset",
+                json={
+                    "email": user.email
+                },
+            )
+
+            # Test that email will get sent
+
+            # self.assertIn("message", resp.get_json())
+            # self.assertEqual(resp.status_code, 200)
+
+    def test_password_reset_token_correct(self):
+        with self.app_context():
+            user = self.create_user()
+            token = user.get_password_reset_token()
+
+            resp = self.client.post(
+                "/reset_password",
+                json={
+                    "token": token,
+                    "password": "testpass",
+                    "password2": "testpass"
+                }
+            )
+
+            self.assertIn("message", resp.get_json())
+            self.assertEqual(resp.status_code, 200)
+            self.assertTrue(user.verify_password("testpass"))
+
+    def test_password_reset_no_match(self):
+        with self.app_context():
+            user = self.create_user()
+            token = user.get_password_reset_token()
+            print(token)
+
+            resp = self.client.post(
+                "/reset_password",
+                json={
+                    "token": token,
+                    "password": "testpass1",
+                    "password2": "testpass2"
+                }
+            )
+
+            self.assertIn("message", resp.get_json())
+            self.assertEqual(resp.status_code, 400)
+
+
+    def test_password_reset_token_expired(self):
+        with self.app_context():
+            user = self.create_user()
+            with freeze_time("2020-04-18 22:10:00"):
+                token = user.get_password_reset_token()
+                print(token)
+
+            # User uses expired token
+            with freeze_time("2020-04-19 22:10:00"):
+                resp = self.client.post(
+                    "/reset_password",
+                    json={
+                        "token": token,
+                        "password": "testpass",
+                        "password": "testpass"
```

Comment:
This does not make sense.

Reply:
That was a typo, will be solved since json now only needs "token" and "password" entries, no "password2".